packages/react/slot/src/Slot.tsx

```diff
@@ -13,28 +13,24 @@ const Slot = React.forwardRef<HTMLElement, SlotProps>((props, forwardedRef) => {
   const { children, ...slotProps } = props;
   const childLength = React.Children.count(children);
 
-  if (childLength === 1) {
-    return (
-      <SlotClone {...slotProps} ref={forwardedRef}>
-        {children}
-      </SlotClone>
-    );
-  }
-
-  return (
+  return childLength === 1 ? (
+    <SlotClone {...slotProps} ref={forwardedRef}>
+      {children}
+    </SlotClone>
+  ) : React.Children.toArray(children).some(isSlottable) ? (
     <>
       {React.Children.map(children, (child) => {
-        if (React.isValidElement(child) && child.type === Slottable) {
-          return (
-            <SlotClone {...slotProps} ref={forwardedRef}>
-              {child.props.children}
-            </SlotClone>
-          );
-        }
-
-        return child;
+        return isSlottable(child) ? (
+          <SlotClone {...slotProps} ref={forwardedRef}>
+            {child.props.children}
+          </SlotClone>
+        ) : (
+          child
+        );
       })}
     </>
+  ) : (
+    <>{React.Children.only(children)}</>
```

Comment:
This is now saying that if the length is greater than `1` and none of the children contain a `Slottable` then it is the consumer code providing more than one child so we throw the `React.Children.only` error.

Reply:
Is it expected that `< 1` should throw? seems that the `WithoutSlottable` story now throws while `WithSlottable` does not, this was the opposite before.

Reply:
If there isn't a `Slottable` it should throw. I've updated the Chromatic stories to reflect the expectation. Let me know if they're clearer now :+1:

Reply:
Thanks for that, it does make more sense now. What I'm still finding confusing though are the `boolean` values as children. Right now `false` won't throw yet `null` will, I kind of expected those to behave consistently?

Reply:
Oh, let me take a look... that might be another bug haha

Reply:
> I think that if the consumer does false && ... it should error because asChild expects a child.

I think this is where I'm getting confused because now, with this change, in collapsible `isOpen && children` wont error (that's the fix) but if I wrote it as `isOpen ? children : null` it would. I know technically why that is, but from a consumer perspective that seems inconsistent, my heads spins a bit with some of the slot stuff though so perhaps I'm not following? 🤔

Reply:
Haha, there is a difference between what the consumer passes and what we do internally. Let's call about this quick so I explain... it is the reason why we have the `Slottable`.

However, I've actually had to revert to allowing the consumer to pass `null` or `false` because I can't seem to determine if a falsey child was provided by the consumer or our code so tests look like this now:

![image](https://user-images.githubusercontent.com/175330/133308890-13f372ae-5558-40da-bce0-2bbc17f3402a.png)

I think this is fine for now... it would have been a nice to have to make `asChild` error if there was no child but it's not the end of the world if it doesn't.